File: dotnet/Library/Okapi/DIDComm.cs
using Okapi.Transport.V1;

namespace Okapi.Transport;

public static class DIDComm
{
    public static PackResponse Pack(PackRequest request)
    {
        return Native.Call<PackRequest, PackResponse>(request, Native.didcomm_pack);
    }

    public static UnpackResponse Unpack(UnpackRequest request)
    {
        return Native.Call<UnpackRequest, UnpackResponse>(request, Native.didcomm_unpack);
    }

    public static SignResponse Sign(SignRequest request)
    {
        return Native.Call<SignRequest, SignResponse>(request, Native.didcomm_sign);
    }

    public static VerifyResponse Verify(VerifyRequest request)
    {
        return Native.Call<VerifyRequest, VerifyResponse>(request, Native.didcomm_verify);
    }
}

File: python/trinsicokapi/didcomm.py
from trinsicokapi.proto.okapi.transport.v1 import (
    VerifyRequest,
    SignRequest,
    UnpackRequest,
    PackRequest,
    PackResponse,
    UnpackResponse,
    SignResponse,
    VerifyResponse,
)
from trinsicokapi.wrapper import _typed_wrap_and_call


def pack(request: PackRequest) -> PackResponse:
    response = _typed_wrap_and_call("didcomm_pack", request, PackResponse)
    return response


def unpack(request: UnpackRequest) -> UnpackResponse:
    response = _typed_wrap_and_call("didcomm_unpack", request, UnpackResponse)
    return response


def sign(request: SignRequest) -> SignResponse:
    response = _typed_wrap_and_call("didcomm_sign", request, SignResponse)
    return response


def verify(request: VerifyRequest) -> VerifyResponse:
    response = _typed_wrap_and_call("didcomm_verify", request, VerifyResponse)
    return response
